plugins/action/system/shutdown.py

```diff
@@ -0,0 +1,234 @@
+# Copyright: (c) 2016-2018, Matt Davis <mdavis@ansible.com>
+# Copyright: (c) 2018, Sam Doran <sdoran@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import random
+import time
+
+from datetime import datetime, timedelta
+
+from ansible.errors import AnsibleError, AnsibleConnectionFailure
+from ansible.module_utils._text import to_native, to_text
+from ansible.module_utils.common.collections import is_string
+from ansible.module_utils.common.validation import check_type_str
+from ansible.plugins.action import ActionBase
+from ansible.utils.display import Display
+
+display = Display()
+
+
+class TimedOutException(Exception):
+    pass
+
+
+class ActionModule(ActionBase):
+    TRANSFERS_FILES = False
+    _VALID_ARGS = frozenset((
+        'connect_timeout',
+        'msg',
+        'delay',
+        'search_paths'
+    ))
+
+    DEFAULT_CONNECT_TIMEOUT = None
+    DEFAULT_PRE_SHUTDOWN_DELAY = 0
+    DEFAULT_SHUTDOWN_MESSAGE = 'Shut down initiated by Ansible'
+    DEFAULT_SHUTDOWN_COMMAND = 'shutdown'
+    DEFAULT_SHUTDOWN_COMMAND_ARGS = '-h {delay_min} "{message}"'
+    DEFAULT_SUDOABLE = True
+
+    DEPRECATED_ARGS = {}
+
+    SHUTDOWN_COMMANDS = {
+        'alpine': 'poweroff',
+        'vmkernel': 'halt',
+    }
+
+    SHUTDOWN_COMMAND_ARGS = {
+        'alpine': '',
+        'void': '-h +{delay_min} "{message}"',
+        'freebsd': '-h +{delay_sec}s "{message}"',
+        'linux': DEFAULT_SHUTDOWN_COMMAND_ARGS,
+        'macosx': '-h +{delay_min} "{message}"',
+        'openbsd': '-h +{delay_min} "{message}"',
+        'solaris': '-y -g {delay_sec} -i 5 "{message}"',
+        'sunos': '-y -g {delay_sec} -i 5 "{message}"',
+        'vmkernel': '-d {delay_sec}',
+        'aix': '-Fh',
+    }
+
+    def __init__(self, *args, **kwargs):
+        super(ActionModule, self).__init__(*args, **kwargs)
+
+    @property
+    def delay(self):
+        return self._check_delay('delay', self.DEFAULT_PRE_SHUTDOWN_DELAY)
+
+    def _check_delay(self, key, default):
+        """Ensure that the value is positive or zero"""
+        value = int(self._task.args.get(key, self._task.args.get(key + '_sec', default)))
```

Comment:
Why `key + '_sec'`?

Reply:
Fixed in https://github.com/ansible-collections/community.general/pull/700/commits/879ce8345257af1c1c25cd46adbfe7b95f2565b6.
Actually I don't even why was the reason for second `self._task.args.get(key, default)`, I checked and they're all the same, therefore I have removed the second `args.get` completely.
I think this is sufficient:
`value = int(self._task.args.get(key, default))`